lib/tidewave/mcp/git_ls.ex

```diff
@@ -12,33 +12,18 @@ defmodule Tidewave.MCP.GitLS do
   end
 
   defp execute_git(fun) do
-    cond do
-      !System.find_executable("git") ->
-        {:error, "This tool requires git to be installed and available in the PATH."}
-
-      File.dir?(".git") ->
-        fun.(nil)
-
-      true ->
-        # create an empty git repo to run ls-files
-        tmp_dir = Path.join(Mix.Project.build_path(), "tmp")
-        git_dir = Path.join(tmp_dir, ".git")
-
-        if !File.dir?(git_dir) do
-          {_, 0} = System.cmd("git", ["init", tmp_dir])
-        end
-
-        fun.(git_dir)
+    if git = MCP.git_root() do
+      fun.(Path.join(git, ".git"))
+    else
```

Comment:
when there's no git root, fetch_env would fail and we would not return an error here, right? So it would need to use get_env instead